### PR TITLE
dev link liquid tag render utilize html.erb

### PR DESCRIPTION
**Pull request by mariocsee**
## What type of PR is this? (check all applicable)

- [x] Refactor

## Description
Refactor how dev ink liquid tag renders to view. 

## Related Tickets & Documents
Tracker #2559 
Dev comment refactor #2554

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
N/A 

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

File: app/views/articles/_liquid.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  <a href='<%= article.user.path %>' class='ltag__link__link'>
+    <div class='ltag__link__pic'>
+      <img src='<%= ProfileImage.new(article.user).get(150) %>' alt='<%= article.user.username + " " + "image"%>'/>
+    </div>
+  </a>
+  <a href='<%= article.path %>' class='ltag__link__link'>
+    <div class='ltag__link__content'>
+      <h2><%= title %></h2>
+      <h3><%= article.user.name %></h3>
+      <div class='ltag__link__taglist'><%= article.tag_list.map { |t| "<span class='ltag__link__tag'>##{t}</span>" }.join.html_safe %></div>
```
Comment (maestromac) [inline app/views/articles/_liquid.html.erb]: Let's not generate html element via string + html_safe here.

File: spec/liquid_tags/link_tag_spec.rb [hunk 2]
```diff
@@ -19,20 +19,21 @@ def generate_new_liquid(slug)
...
-          </div></a>
-          <a href='#{article.path}' class='ltag__link__link'>
-            <div class='ltag__link__content'>
-              <h2>#{ActionController::Base.helpers.strip_tags(article.title)}</h2>
-              <h3>#{article.user.name}</h3>
-              <div class='ltag__link__taglist'>#{tags}</div>
-            </div>
-        </a>
-      </div>
+    <<~HTML
```
Comment (maestromac) [inline spec/liquid_tags/link_tag_spec.rb]: nit: Something happened to the spacing here?

File: app/liquid_tags/link_tag.rb [hunk 3]
```diff
@@ -1,52 +1,32 @@
 class LinkTag < LiquidTagBase
   include ActionView::Helpers
-  attr_reader :article
+  PARTIAL = "articles/liquid".freeze
 
   def initialize(_tag_name, slug_or_path_or_url, _tokens)
-    @article = parse_url_for_article(slug_or_path_or_url)
+    @article = get_article(slug_or_path_or_url)
+    @title = strip_tags @article.title
```
Comment (maestromac) [inline app/liquid_tags/link_tag.rb]: nit: while we don't have a strict rule, it's a good idea to stay consistent.
```suggestion
    @title = strip_tags(@article.title)
```

File: spec/liquid_tags/link_tag_spec.rb (hunk 2 above)
Comment (mariocsee) [reply spec/liquid_tags/link_tag_spec.rb]: fixed spacing in recent commit

File: app/liquid_tags/link_tag.rb (hunk 3 above)
Comment (mariocsee) [reply app/liquid_tags/link_tag.rb]: updated with your suggested change.

File: app/views/articles/_liquid.html.erb (hunk 1 above)
Comment (mariocsee) [reply app/views/articles/_liquid.html.erb]: generating html through `<% article.tag_list.each do |t| %> <span ... > <%= t %> </span><% end %>` in recent commit

**Pull request merged**